File: scripts/overrides/item_price.py
import frappe
from erpnext.stock.doctype.item_price.item_price import ItemPrice


class CustomItemPrice(ItemPrice):
	def validate_item_template(self):
		"""Allow creating Item Price for template items.

		ERPNext v15+ blocks Item Prices for template items (items with
		``has_variants = 1``). This override removes that restriction so
		you can set a single price on the template and all variants will
		inherit it automatically.

		The variant→template price fallback already exists in
		``erpnext.stock.get_item_details.get_price_list_rate`` (line ~1043),
		so no additional changes are needed on the fetching side.
		"""
		pass

File: scripts/public/js/item_price.js
/**
 * Item Price – allow selecting template items (has_variants = 1).
 *
 * ERPNext v15+ filters out template items in the item_code link field
 * (has_variants: 0). This override removes that filter so you can set
 * prices on template items, which variants will then inherit.
 */
frappe.ui.form.on("Item Price", {
	setup: function (frm) {
		// Remove the has_variants: 0 filter that blocks template items.
		frm.set_query("item_code", function () {
			return {};
		});
	},
});
